addons/l10n_ro_edi_stock/models/l10n_ro_edi_stock_etransport_interface.py

```diff
@@ -0,0 +1,670 @@
+            return errors  # return prematurely because all the end location fields depend on this field
+
+        # Location fields
+        for location in ('start', 'end'):
+            loc_value = self[f'l10n_ro_edi_stock_{location}_loc_type']
+            loc_group = _("'Start Location'") if location == 'start' else _("'End Location'")
+
+            if loc_value == 'bcp' and not self[f'l10n_ro_edi_stock_{location}_bcp']:
+                errors.append(_("The border crossing point is missing under %(location_group)s", location_group=loc_group))
+            elif loc_value == 'customs' and not self[f'l10n_ro_edi_stock_{location}_customs_office']:
+                errors.append(_("The customs office is missing under %(location_group)s", location_group=loc_group))
+            elif loc_value == 'location':
+                missing_field_names = []
+                if not self[f'l10n_ro_edi_stock_{location}_state_id']:
+                    missing_field_names.append(_("State"))
+                if not self[f'l10n_ro_edi_stock_{location}_city']:
+                    missing_field_names.append(_("City"))
+                if not self[f'l10n_ro_edi_stock_{location}_street']:
+                    missing_field_names.append(_("Street"))
+                if not self[f'l10n_ro_edi_stock_{location}_zip']:
+                    missing_field_names.append(_("Postal Code"))
+
+                if len(missing_field_names) == 1:
+                    errors.append(_("%(location_group)s is missing the %(field_name)s field.", location_group=loc_group, field_name=missing_field_names[0]))
+                elif len(missing_field_names) > 1:
+                    errors.append(_("%(location_group)s is missing following fields: %(field_names)s", location_group=loc_group, field_names=missing_field_names))
+
+        return errors
+
+    def _check_fetch_values(self, errors=None) -> list:
+        if errors is None:
+            errors = []
+        self.ensure_one()
+
+        if not self.l10n_ro_edi_stock_company_id.l10n_ro_edi_access_token:
+            errors.append(_('Romanian access token not found. Please generate or fill it in the settings.'))
+            return errors
+
+        match self.l10n_ro_edi_stock_state:
+            case 'stock_sending_failed':
+                if not self._get_last_stock_validated_document():
+                    errors.append(_("This document has not been successfully sent yet because it contains errors."))
+                else:
+                    errors.append(_("This document has not been corrected yet because it contains errors."))
+            case 'stock_validated':
+                errors.append(_("This document has already been successfully sent to anaf."))
+
+        return errors
+
+    ################################################################################
+    # Template Helpers
+    ################################################################################
+
+    def _get_declarant_ref(self) -> str:
```

Comment:
All those getters are useless. Just make a method returning all the values to render the xml and make all the computation inside.